tdrs-backend/tdpservice/data_files/admin/admin.py

```diff
@@ -23,11 +24,45 @@ def has_change_permission(self, request, obj=None):
         """Read only permissions."""
         return False
 
-
 @admin.register(DataFile)
 class DataFileAdmin(ReadOnlyAdminMixin, admin.ModelAdmin):
     """Admin class for DataFile models."""
 
+    class Media:
+        """Media class for DataFileAdmin."""
+
+        js = ('admin/js/admin/mymodel.js',)
+
+    actions = ['reparse_cmd']
+
+    def reparse_cmd(self, request, queryset):
+        """Reparse the selected data files."""
+        files = queryset.values_list("id", flat=True)
+        file_ids = ",".join(map(str, files))
+        from django.core.management import call_command
+        from django.utils.translation import ngettext
+        from django.contrib import messages
+        call_command("clean_and_reparse", f"-f {file_ids}")
```

Comment:
this works, but my preference would be the re-implement the logic of the management command here

Reply:
one reason for this - i was testing the sequential reparse scenario and was confronted with this error

![image](https://github.com/user-attachments/assets/1c0db850-d9a1-4cbc-8150-e00787cabf3d)

i couldn't find the actual exception in the logs (the parse was still active, and putting out tons of logs), but if we re-implemented the command logic, that error could be caught and displayed to the user

Reply:
You should be able to catch the exception/log message in the admin console LogEntries. It should make a new entry indicating it wont reparse when not sequential

Reply:
I also agree with Jan that as a first pass calling the command works, but it would be nice to share the logic between this and the command to allow for a little more fine grain control.

Reply:
so are we ok to repeat the steps?

Reply:
I don't think they need to have code duplicates. I bet we can factor out parametrized functions from the command and put them into a shared module for both the command and the action to import. That will be a good amount of work though which I think could make sense to move into a follow on ticket.

Reply:
we can make a util file and create util functions and share them and I agree this can be a new ticket

Reply:
Created a new issue #3205